src/Durable/DurableController.cs

```diff
@@ -0,0 +1,143 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+namespace Microsoft.Azure.Functions.PowerShellWorker.Durable
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Linq;
+    using System.Management.Automation;
+
+    using Newtonsoft.Json;
+    using WebJobs.Script.Grpc.Messages;
+
+    using PowerShellWorker.Utility;
+
+    /// <summary>
+    /// The main entry point for durable functions support.
+    /// </summary>
+    internal class DurableController
+    {
+        private readonly bool _durableEnabled;
+        private readonly DurableFunctionInfo _durableFunctionInfo;
+        private readonly IPowerShellServices _powerShellServices;
+        private readonly IOrchestrationInvoker _orchestrationInvoker;
+        private OrchestrationBindingInfo _orchestrationBindingInfo;
+
+        public DurableController(
+            DurableFunctionInfo durableDurableFunctionInfo,
+            PowerShell pwsh)
+            : this(
+                Utils.AreDurableFunctionsEnabled(),
+                durableDurableFunctionInfo,
+                new PowerShellServices(pwsh),
+                new OrchestrationInvoker())
+        {
+        }
+
+        internal DurableController(
+            bool durableEnabled,
+            DurableFunctionInfo durableDurableFunctionInfo,
+            IPowerShellServices powerShellServices,
+            IOrchestrationInvoker orchestrationInvoker)
+        {
+            _durableEnabled = durableEnabled;
+            _durableFunctionInfo = durableDurableFunctionInfo;
+            _powerShellServices = powerShellServices;
+            _orchestrationInvoker = orchestrationInvoker;
+        }
+
+        public void BeforeFunctionInvocation(IList<ParameterBinding> inputData)
+        {
+            // If the function is an orchestration client, then we set the OrchestrationClient
+            // in the module context for the 'Start-NewOrchestration' function to use.
+            if (_durableFunctionInfo.IsOrchestrationClient)
+            {
+                ThrowIfDurableNotEnabled();
```

Comment:
Does this method get called for non durable functions? If the answer is no, then would it be possible to move `ThrowIfDurableNotEnabled();` outside the branches? e.g., 

```
public void BeforeFunctionInvocation(IList<ParameterBinding> inputData)
{
   ThrowIfDurableNotEnabled();
   ...
```

Reply:
It actually does get called for all functions, including non-durable. I'm intentionally moving checking function type, checking the feature flag, and throwing the exception into the Durable folder/namespace. This logic is small and simple, but it already belongs to Durable.